media_utils/video_puppeteer.py

```diff
@@ -150,49 +138,29 @@ class VideoException(Exception):
 
 def playback_started(video):
     try:
-        return video.current_time > 0
+        return video.current_time > video.start_time
     except Exception as e:
         print ('Got exception %s' % e)
         return False
 
-# Debug info looks something like:
-# Dumping data for reader d10c000:
-#  Dumping Audio Track Decoders: - mLastAudioTime: 9.386666
-#   Reader 0: d2df400 ranges=[(0.000000, 16.042666)] active=true size=196456
-#  Dumping Video Track Decoders - mLastVideoTime: 8.466791
-#	 Reader 2: 13f54400 ranges=[(8.008000, 16.016000)] active=true size=3741917
-#	 Reader 1: d2edc00 ranges=[(4.004000, 8.008000)] active=false size=1088692
-#	 Reader 0: d2eac00 ranges=[(2.002000, 4.004000)] active=false size=242367
-
-# One simple hack: There should be two active readers at any one time. If
-# there are not exactly two, something is horribly wrong.
-
 
 def playback_done(video):
-    active_re = re.compile('active=true', re.DOTALL)
-
     # If we are near the end and there is still a video element, then
-    # we are essentially done. Any further progress might set on of
-    # the stream active state to false, and we will raise in that
-    # case inappropriately.
+    # we are essentially done. If this happens to be last time we are polled
+    # before the video ends, we won't get another chance.
 
-    if abs(video.remaining_time) < 2.0:
+    remaining_time = video.remaining_time
```

Comment:
I tried that. I think interval was too small. I did try it and it did not work well.